lib/reservoir-sampler.js

```diff
@@ -89,17 +103,26 @@ ReservoirSampler.prototype._read = function() {
     debug('sampling %d documents from a collection with %d documents',
       this.size, count);
 
+    var minSize = Math.min(this.size, count, RESERVOIR_SAMPLE_LIMIT);
```

Comment:
This definition of "minSize", and how it's used below as the "size" argument to batchDocs(), is interesting. Currently, I believe this implies: the # and size of batches is not simply dependent on the desired sample size. It also depends on the *expected* collection size.

So if we want a sample of 1000 docs but the collection contains 80 docs, we'll make 9 separate batches of about 8 documents? This is a little hard to reason about.

Is `count` ever stale? If it was, (e.g. we obtained a collection count some time ago, and documents were inserted/removed in the meantime), the sample may be unnecessarily small.

For ease of explaining how sampling works, I would prefer that the batch size depends only on the desired sample size - and thus independent of `count`.